docs/features/optin.md

```diff
@@ -0,0 +1,104 @@
+# Opt in to Enable Experimental Direct Controllers
+
+> `Feature State`: `alpha` as of version v1.125
+
+KCC can be configured to enable experimental versions of direct controllers for
+reconciling specific resources. This will allow users to test the new direct
+controller code for bug fixes or features not available in legacy controllers.
+It is possible to enable the direct controller code on a per-resource basis,
+by setting an annotation for each custom resource.
+
+
+## Enabling
+
+To enable the direct controller for a specific resource, update the resource and
+specify the annotation:
+
+```yaml
+metadata:
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+```
+
+For example:
+
+```yaml
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: my-sqlinstance
+  annotations:
+    alpha.cnrm.cloud.google.com/reconciler: "direct"
+spec:
+  databaseVersion: POSTGRES_16
+  region: us-central1
+  settings:
+    tier: db-custom-1-3840
+```
+
+Note: The only supported value for the annotation is "direct", to enable the
+direct controller. To revert back to the legacy controller, remove the
+annotation.
+
+Note, it's harmless to add the annotation to enable experimental direct
+controllers for resources that do not support an experimental direct controller.
+In this case, the normal (default) controller will be used.
+
+
+## Verifying
+
+To verify the direct controller code is in-use for a particular resource,
+check the logs for the phrase "Running reconcile" associated with the resource.
+
+```
+$ kubectl -n cnrm-system logs pod/cnrm-controller-manager-0 | grep -e "Running reconcile" -e "my-sqlinstance"
+```
+
+You should see output similar to the following:
+
+> {"severity":"info","timestamp":"2024-11-04T21:56:28.704Z","msg":"Running reconcile","controller":"sqlinstance-controller","controllerGroup":"sql.cnrm.cloud.google.com","controllerKind":"SQLInstance","SQLInstance":{"name":"my-sqlinstance","namespace":"default"},"namespace":"default","name":"my-sqlinstance","reconcileID":"577fe58d-14eb-4e5d-9642-c9ec4b1a3137","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+If the legacy controller is enabled (instead of the new direct controller), the
+logs will show "starting reconcile" for the resource instead of "Running reconcile".
+
+> {"severity":"info","timestamp":"2024-10-30T23:47:49.726Z","logger":"sqlinstance-controller","msg":"starting reconcile","resource":{"name":"my-sqlinstance","namespace":"default"}}
+
+
+## Applicability
+
+We recommend only enabling the experimental direct controller if it is
+necessary to work around a legacy controller bug or enable a new feature that
+is only available in the direct controller.
```

Comment:
I like this statement. Can we move it to the top before "enabling", in case users turn it on unexpected before seeing the statement.

Reply:
Done